src/main/java/net/minestom/server/network/packet/server/play/JoinGamePacket.java

```diff
@@ -54,12 +87,23 @@ public void write(@NotNull NetworkBuffer writer) {
         //is flat
         writer.write(BOOLEAN, isFlat);
 
-        writer.write(BOOLEAN, false);
+        if (deathLocation() == null) {
+            writer.write(BOOLEAN, false);
+        }
+        else {
+            writer.write(BOOLEAN, true);
+            writer.write(STRING, deathLocation().dimension());
+            writer.write(BLOCK_POSITION, deathLocation().position());
+        }
     }
 
     @Override
     public int getId() {
         return ServerPacketIdentifier.JOIN_GAME;
     }
 
+    public record DeathLocation(@NotNull String dimension, @NotNull Point position) {
```

Comment:
I believe that the same is present in the respawn packet

Reply:
Just looked at the protocol spec and you're correct. I'm thinking then that `DeathLocation` should be relocated into the `net.minestom.server.network.packet.server.play.data` package. Are these two instances enough to warrant a `NetworkBufferType`, or would a static parser be enough?

Reply:
A new network type seems fine, we can always remove it down the line

Reply:
Working on it now... I don't really know what all the `return -1;` signify in regards to the type writers though.

Reply:
This is used to signed the buffer how much to advance the writer index, `-1` indicate nothing/ignore as using the `write` method inside the lambda already advance the index